### PR TITLE
Tm 668 implement footer aggregate view for terrafund dashboard

### PR DESCRIPTION
**Pull request by LimberHope**
[task link](https://gfw.atlassian.net/browse/TM-668)

### CODE REVIEW

File: app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    public function __invoke()
+    {
+        return response()->json([
+            'active_countries' => $this->getAllContries(),
+        ]);
+    }
+
+    public function getAllContries()
+    {
+        $conuntryId = FormOptionList::where('key', 'countries')->value('id');
```
Comment (pachonjcl) [inline app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php:25]: fix this typo

File: app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        return Project::where('framework_key', 'terrafund')
+            ->where('country', $country);
+    }
+
+    public function numberOfProjects($country)
+    {
+        $totalNumberOfProjects = $this->projectsCountry($country)
+                ->count();
+
+        return $totalNumberOfProjects;
```
Comment (pachonjcl) [inline app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php:65]: If you are going to just return it, there is no need to store in a intermediate variable.

File: app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    public function numberOfNurseries($country)
+    {
+        $projects = $this->projectsCountry($country)
+            ->get();
+        $totalNurseries = 0;
+        foreach ($projects as $project) {
+            $totalNurseries += $project->nurseries()->count();
+        }
+
+        return $totalNurseries;
```
Comment (pachonjcl) [inline app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php:124]: I just noted that you are using the `foreach` method only for the sake of summing things.

Can you take a look at [documentation](https://laravel.com/docs/10.x/collections#method-sum) and see what is the best way to use in the endpoints?

**Review by pachonjcl — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php (hunk 3 above)
Comment (LimberHope) [reply app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php:124]: I have been looking the documentation and the best option is to use the SUM method, now i upload the changes.

File: app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    public function projectsCountry($country)
+    {
+        return Project::where('framework_key', 'terrafund')
+            ->where('country', $country);
+    }
+
+    public function numberOfProjects($country)
+    {
+        return $this->projectsCountry($country)
+                ->count();
```
Comment (pachonjcl) [inline app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php:63]: Keep It one line.

File: app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    public function totalSpeciesAmount($country)
+    {
+        $projects = $this->projectsCountry($country)
+            ->get();
```
Comment (pachonjcl) [inline app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php:69]: one line too.

File: app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            }
+        }
+
+        return $totalSpeciesAmount;
+    }
+
+    public function totalJobsCreated($country)
+    {
+        $projects = $this->projectsCountry($country)
+            ->get();
```
Comment (pachonjcl) [inline app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php:89]: same here.

**Review by pachonjcl — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+    public function totalJobsCreated($country)
+    {
+        $projects = $this->projectsCountry($country)->get();
+        $totalJobs = 0;
+        foreach ($projects as $project) {
+            $latestProjectReport = ProjectReport::where('project_id', $project->id)
+                ->orderByDesc('due_at')
+                ->value(DB::raw('pt_total + ft_total'));
+            $totalJobs += $latestProjectReport;
+        }
```
Comment (pachonjcl) [inline app/Http/Controllers/V2/Dashboard/ActiveCountriesTableController.php:93]: replace it with a `sum`

**Review by pachonjcl — CHANGES_REQUESTED** [review]

**Pull request merged**